eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -311,42 +312,53 @@ private[journal] object ReplicatedCassandra {
                               .as(journalHead.some)
                           } { journalHead =>
                             if (offset > journalHead.partitionOffset.offset) {
-                              val deleteTo0 = journalHead.deleteTo
-                              val seqNr     = journalHead.seqNr
-                              val seqNr1    = deleteTo0.fold { seqNr } { _.value.max(seqNr) }
-                              val deleteTo1 = deleteTo
+                              val headDeleteTo = journalHead.deleteTo
+                              val headSeqNr    = journalHead.seqNr
+
+                              // keep existing `seqNr` or update to higher one
+                              // usually because `delete batch` removed previous `append batch`
+                              val validatedSeqNr: SeqNr = setSeqNr.fold { headSeqNr } { _.max(headSeqNr) }
+
+                              // clamp down to requested or max valid seqNr
+                              // in default protocol we have `Journal#delete(DeleteTo.max)`
+                              // `DeleteTo.max` can be used to discard all traces, but keep `seqNr`
+                              val validatedDeleteTo = deleteTo
                                 .value
-                                .min(seqNr1)
+                                .min(validatedSeqNr)
                                 .toDeleteTo
+
+                              val updateSeqNr = setSeqNr.isDefined
+                              // different from value in head, must be less or equal to head's `seqNr`
+                              val updateDeleteTo = headDeleteTo.forall(_ < validatedDeleteTo)
```

Comment:
Before the change, `delete_to` was updated if new value does not equal to new one:
```
                                    if (deleteTo0.contains(deleteTo1)) {
                                      update()
                                    } else {
                                      update(deleteTo1)
                                    }
```
or
```
                                    if (deleteTo0.contains(deleteTo1)) {
                                      update(seqNr1)
                                    } else {
                                      update(seqNr1, deleteTo1)
                                    }
```
while you changed it on more loose condition:
```
headDeleteTo.forall(_ < validatedDeleteTo)
```

Reply:
Unwrapped "before":
```scala
if ( ! journalHead.deleteTo.contains(deleteTo.value.min(journalHead.seqNr).toDeleteTo) ) {
// update `delete_to` fiels
}
```

Like "if head's delete_to is empty or doesn't contain min(requested or head's seq_nr)".

Unwrapped in this PR:
```scala
val s = setSeqNr.fold { journalHead.seqNr } { _.max(journalHead.seqNr) }
if (journalHead.deleteTo.forall(_ < deleteTo.value.min(s).toDeleteTo)) {
// update `delete_to` fiels
}
```

Now "if head's delete_to is empty of the existing value is smaller than min(requested or max(head's seq_nr or <from removed append action> ))"

Reply:
could you please explain why before `delete_to` would be updated even if 
```
deleteTo (incoming from an app) > meta.seq_nr > meta.delete_to
```